### PR TITLE
Mm 38591 allow disabling status update

### PR DESCRIPTION
**Pull request by gbochora**
#### Summary
This pull request will introduce a per-playbook configuration that allows disabling the status update functionality for subsequent runs created using that playbook.

- Added **Enable status updates** toggle to playbook edit form. When disabled, **Default update timer** and **Status updates** configurations for a playbook are disabled. Additionally, the **When an update is posted** actions are disabled altogether:

https://user-images.githubusercontent.com/4368372/143412197-6d4b47e9-225a-435e-a06b-fbf9bc41ae9a.mp4

- Removed status update section on Playbook preview form, when status updates are disabled

https://user-images.githubusercontent.com/4368372/143412407-ffe75c26-d30f-4068-9334-2264874e2342.mp4

- Removed RHS post update section, when status updates are disabled

https://user-images.githubusercontent.com/4368372/143412588-c24c9731-6448-409f-b63b-c9981d861167.mp4

- Added special message to Playbook details form, when status updates are disabled

https://user-images.githubusercontent.com/4368372/143412712-61bec00c-633d-4061-8cb5-078beb96e377.mp4

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38591

#### Checklist

~~- [ ] Telemetry updated~~
~~- [ ] Gated by experimental feature flag~~
- [ ] Unit tests updated

### CODE REVIEW

File: tests-e2e/cypress/integration/playbooks/edit_spec.js [hunk 1]
```diff
@@ -1195,5 +1195,173 @@ describe('playbooks > edit', () => {
                 });
             });
         });
+
+        describe('status updates enable / disabled', () => {
```
Comment (cpoile) [inline tests-e2e/cypress/integration/playbooks/edit_spec.js:1199]: This is amazing e2e work, thank you!

**Review by cpoile — APPROVED** [review]
This is great, thanks @gbochora!
Sending it to Alejandro for one more pair of eyes.

**Comment by cpoile** [comment]
@gbochora Oh, there's a conflict with the e2e files on master, just fyi.

File: tests-e2e/cypress/integration/channels/rhs/status_update_spec.js [hunk 2]
```diff
@@ -242,4 +242,30 @@ describe('channels > rhs > status update', () => {
...
+                title: 'Playbook',
+                userId: testUser,
+                broadcastChannelId: testChannel.id,
+                statusUpdateEnabled: false,
+            }).then((playbook) => {
+                testPlaybook = playbook;
+            });
+        });
+
+        describe('Omit status update dialog when status updates ar disabled', () => {
```
Comment (agarciamontoro) [inline tests-e2e/cypress/integration/channels/rhs/status_update_spec.js:260]: Typo (and I think we always start with lowercase):
```suggestion
        describe('omit status update dialog when status updates are disabled', () => {
```

File: tests-e2e/cypress/integration/channels/rhs/status_update_spec.js [hunk 3]
```diff
@@ -242,4 +242,30 @@ describe('channels > rhs > status update', () => {
...
+            it('shows the default when we have not made an update before', () => {
+                // # Check if RHS section is loaded
+                cy.get('#rhs-about').should('exist');
+
+                // # Check if Post Update section is omitted
+                cy.get('#rhs-post-update').should('not.exist');
```
Comment (agarciamontoro) [inline tests-e2e/cypress/integration/channels/rhs/status_update_spec.js:266]: The convention we follow in the E2E tests is to use an asterisk for verification steps:

```suggestion
                // * Check if RHS section is loaded
                cy.get('#rhs-about').should('exist');

                // * Check if Post Update section is omitted
                cy.get('#rhs-post-update').should('not.exist');
```

File: tests-e2e/cypress/integration/runs/details_spec.js [hunk 4]
```diff
@@ -108,4 +108,40 @@ describe('runs > details', () => {
...
+                    playbookRunName: name,
+                    ownerUserId: testUser.id,
+                });
+            }).then((run) => {
+                playbookRun = run
+            })
+        });
+
+
+        it('should shows, that status updates were disabled', () => {
```
Comment (agarciamontoro) [inline tests-e2e/cypress/integration/runs/details_spec.js:139]: Typo
```suggestion
        it('should show that status updates were disabled', () => {
```

**Review by agarciamontoro — APPROVED** [review]
This is great, @gbochora, thank you! :tada:

I left a couple of nits and typos, but nothing blocking, so I'm approving now!

One thing we will need to do in a future PR is to improve the user feedback on why the actions related to status updates are disabled when the status updates are disabled, as I think it may be a bit confusing right now.

**Comment by agarciamontoro** [comment]
Thanks again, @gbochora! I'll merge as soon as CI passes :)

**Comment by gbochora** [comment]
@cpoile @agarciamontoro  Thanks for the feedback : )

**Pull request merged**